std/range/primitives.d

```diff
@@ -227,6 +227,22 @@ enum bool isInputRange(R) =
     }
     static assert(!isInputRange!VoidFront);
 }
+// Verify fix for: https://issues.dlang.org/show_bug.cgi?id=16034
+@safe unittest
+{
+    struct One
+    {
+        int entry = 1;
+        @disable this(this);
+    }
+
+    One[] ones = [One(), One()];
+
+    import std.algorithm.iteration : map;
+    import std.algorithm.comparison : equal;
+
+    assert(ones.map!`a.entry + 1`.equal([2, 2]));
```

Comment:
Since this is really a test for `isInputRange`, not `map`, I would change this to `static assert(isInputRange!(Ones[]));`.

Reply:
Perhaps I could keep both? As it stands it prevents changes in map that break being able to use map "via a reference only".

Reply:
Keeping both is fine. In that case, let's move this test to `map`'s test suite in `std.algorithm.iteration`.